introduce allow_mutation Field constraint (#2196)
samuelcolvin/pydantic: docs/usage/schema.md

  JSON Schema
* `max_length`: for string values, this adds a corresponding validation and an annotation of `maxLength` to the
  JSON Schema

> * `read_only`: a boolean which defaults to False. When True, the field raises a TypeError if the field is 

Reviewer: The name is misleading, (which I think is why the schema problem has come up). 

The function is not "read only" since the field can be set when creating a model. The name should be `allow_mutation` with default `True`. This is both a better description, and matches `Config.allow_mutation`.

Author: I agree allow_mutation is a better fit and less confusing. Will commit the changes shortly.

Author: @samuelcolvin - how do you feel about the TypeError being raised?  I'm open to other suggestions if you have one.

> * `allow_mutation`: a boolean which defaults to True. When False, the field raises a TypeError if the field is 
>   assigned on an instance.  The BaseModel Config must set validate_assignment to True

Reviewer: ```suggestion
* `allow_mutation`: a boolean which defaults to `True`. When False, the field raises a `TypeError` if the field is 
  assigned on an instance.  The model config must set `validate_assignment` to `True` for this check to be performed.
```
